src/nvim/runtime.c

```diff
@@ -405,31 +415,39 @@ static int add_pack_dir_to_rtp(char_u *fname)
 /// Load scripts in "plugin" and "ftdetect" directories of the package.
 static int load_pack_plugin(char_u *fname)
 {
-  static const char *plugpat = "%s/plugin/**/*.vim";  // NOLINT
+  static const char *plugpat_vim = "%s/plugin/**/*.vim";  // NOLINT
+  static const char *plugpat_lua = "%s/plugin/**/*.lua";  // NOLINT
   static const char *ftpat = "%s/ftdetect/*.vim";  // NOLINT
 
   int retval = FAIL;
   char *const ffname = fix_fname((char *)fname);
   size_t len = strlen(ffname) + STRLEN(ftpat);
-  char_u *pat = try_malloc(len + 1);
-  if (pat == NULL) {
+  char_u *pat_vim = try_malloc(len + 1);
```

Comment:
can use `xmallocz(len)` and skip the NULL checks.

Reply:
`xmallocz` Looks right . One thing though currently `NULL` is recived ffname is freed (threadend at line 453) . We can get rid of that too . NULL means we are out of memory and are crashing . Those leaked memory will be reclaimed by system when the prossess exits . So freeing isn't nessary right ?

Reply:
Also currently `FAIL` is returned only when those memory allocation fails . If those are that's handled by `xmalloc` then return value of this4 function has no significance as it'll always return `OK` and when memory allocation fails nvim get closed . 

Is current setup some how allow to survive out of memory error ?

Reply:
@bfredl updated. Now using `xmallocz`